### DISCUSSION TITLE
Set Page Mode in pypdf 4.1.0

### DISCUSSION THREAD

**Discussion by greg2106** [py-pdf/pypdf#2567] [Q&A]
## Explanation
In PyPDF2 I could simply set the page mode like this:
```
pdf = PyPDF2.PdfFileWriter()
pdf.setPageMode('/UseOutlines')
```

In pypdf there is only:
```
pdf = pypdf.PdfWriter()
pdf._set_page_layout('/SinglePage')
```
(defined in line 2099 in the "_writer.py" source file)

but 
`pdf._set_page_mode('/UseOutlines')`
seems to be missing (if I take the old file structure of PyPDF2 as a basis, I would expect the def in line 2204 of the "_writer.py" source file).

Instead if I want to set the page mode, I have to do:
```
pdf = pypdf.PdfWriter()
pdf.page_mode = '/UseOutlines'
```

Is this intended?

**Comment by stefan6419846** [comment]
Please have a look at the docs: https://pypdf.readthedocs.io/en/stable/user/migration-1-to-2.html Yes, this is correct - using properties instead of getters/setters is the Pythonic way of doing things like this. At some places, this might not be consistent nevertheless.

By the way: Do not use internal APIs in your own code. This might break at any time.
